### PR TITLE
Address clippy suggestions

### PR DESCRIPTION
**Pull request by kov**
I mostly made these mistakes:
  - being too eager to call .clone()
  - using *_or() rather than *_or_else() for defaults from function calls
    which then end up getting called on all cases
  - using closures on .map() to call a single function on the value, or to
    just return the value (doh); in my defense the latter were bad copy/pastes

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![warn(clippy::all)]
```
Comment (scorphus) [inline src/main.rs:1]: I think this attribute is redundant. As per the [docs](https://github.com/rust-lang/rust-clippy/blob/b4e4fa51ea328adf5b182d3820dca9286d5ef3bf/README.md#clippy):
> * `clippy::all` (everything that is on by default: all the categories below except for `nursery`, `pedantic`, and `cargo`)

```suggestion
```

**Review by scorphus — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (kov) [reply src/main.rs:1]: You are right that it is unnecessary for running clippy, but adding it explicitly turns on clippy notes for IDEs like VS Code (if you use the default of opt-in), so I'd prefer to keep it.

Comment (scorphus) [reply src/main.rs:1]: Fair enough 👍 That simply works.

You know.. it seldom feels right to have these "breadcrumbs" of tooling scattered around the code. I did find a couple ways to configure vscode but alas, they're kinda clumsy.

**Review by scorphus — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (kov) [reply src/main.rs:1]: Yeah, not a big fan myself. I'm more ok with clippy I guess as you could consider it rust infrastructure, and explicit being better than implicit and all that but I'd rather it be something we could keep on that multi-IDE configuration file you mentioned hehe

**Pull request merged**